gtclang/test/atlas-integration-test/AtlasIntegrationTestCompareOutput.cpp

```diff
@@ -0,0 +1,196 @@
+//===--------------------------------------------------------------------------------*- C++ -*-===//
+//                         _       _
+//                        | |     | |
+//                    __ _| |_ ___| | __ _ _ __   __ _
+//                   / _` | __/ __| |/ _` | '_ \ / _` |
+//                  | (_| | || (__| | (_| | | | | (_| |
+//                   \__, |\__\___|_|\__,_|_| |_|\__, | - GridTools Clang DSL
+//                    __/ |                       __/ |
+//                   |___/                       |___/
+//
+//
+//  This file is distributed under the MIT License (MIT).
+//  See LICENSE.txt for details.
+//
+//===------------------------------------------------------------------------------------------===//
+#include <fstream>
+#include <gridtools/clang_dsl.hpp>
+
+#include "AtlasVerifyer.h"
+#include "atlas/functionspace/CellColumns.h"
+#include "atlas/functionspace/EdgeColumns.h"
+#include "atlas/grid.h"
+#include "atlas/mesh/actions/BuildCellCentres.h"
+#include "atlas/mesh/actions/BuildEdges.h"
+#include "atlas/meshgenerator.h"
+#include "atlas/option/Options.h"
+#include "atlas/output/Gmsh.h"
+#include "interface/atlas_interface.hpp"
+
+#include <gtest/gtest.h>
+
+#include <generated_copyCell.hpp>
+namespace {
+TEST(AtlasIntegrationTestCompareOutput, CopyCell) {
+  atlas::StructuredGrid structuredGrid = atlas::Grid("L32x32");
+  atlas::StructuredMeshGenerator generator;
+  auto mesh = generator.generate(structuredGrid);
+
+  size_t nb_levels = 1;
+
+  atlas::functionspace::CellColumns fs_cells(mesh, atlas::option::levels(nb_levels));
+  atlas::Field in{fs_cells.createField<double>(atlas::option::name("in"))};
+  atlas::Field out{fs_cells.createField<double>(atlas::option::name("out"))};
+
+  atlasInterface::Field<double> in_v = atlas::array::make_view<double, 2>(in);
+  atlasInterface::Field<double> out_v = atlas::array::make_view<double, 2>(out);
+
+  for(int cell_idx = 0; cell_idx < mesh.cells().size(); ++cell_idx)
+    in_v(cell_idx, 0) = 1.0;
+
+  dawn_generated::cxxnaiveico::copyCell<atlasInterface::atlasTag>(mesh, static_cast<int>(nb_levels),
+                                                                  in_v, out_v)
+      .run();
+
+  for(int cell_idx = 0; cell_idx < mesh.cells().size(); ++cell_idx)
+    ASSERT_EQ(out_v(cell_idx, 0), 1.0);
+}
+} // namespace
```

Comment:
why is each test in it's own private namespace?

Reply:
To not pollute the global namespace. Also need to close it and open again to have the test-specific includes near the test.